### PR TITLE
fix(journeys): fix stages, journey restarts & config, policy display, remove restart after logout

### PR DESCRIPTION
**Pull request by cerebrl**
## Summary

This PR focuses on bug fixes and improvements throughout the journey portion of functionality.

## Details

1. Allows for journey config to be partially configured by dev, rather than all or nothing
2. It removes the auto restart after logout as it was unnecessary and resulting in irrelevant journeys rendering
3. Fixes journey restart after failure by using the journey stack; original attempt at fixing didn't take into account journey switching
4. Renames stages for login and registration to be "DefaultLogin" and "DefaultRegistration", rather than "UsernamePassword" and "Registration", respectively
5. Changes the display of policy rules for callback components as default to opt-in

### CODE REVIEW

**Comment by ryanbas21** [comment]
Your preview environment [pr-92-ryanbas21](https://cloud.okteto.com/#/previews/pr-92-ryanbas21) has been deployed.

  Preview environment endpoint is available [here](https://preview-pr-92-ryanbas21.cloud.okteto.net)

File: src/lib/journey/stages/_utilities/stage.utilities.ts [hunk 1]
```diff
@@ -1,5 +1,4 @@
 import { get } from 'svelte/store';
-import type { z } from 'zod';
```
Comment (cerebrl) [inline src/lib/journey/stages/_utilities/stage.utilities.ts:2]: This was not being used, so just removing.

File: src/lib/journey/stages/index.ts
Comment (cerebrl) [inline src/lib/journey/stages/index.ts:1]: This was not being used anywhere, so I removed it.

File: src/lib/journey/stages/login.svelte [hunk 2]
```diff
@@ -48,9 +48,6 @@
       form?.submit();
     }
   }
-  function returnCallback(callback: any) {
```
Comment (cerebrl) [inline src/lib/journey/stages/login.svelte:51]: Not used.

File: src/lib/journey/stages/registration.svelte [hunk 3]
```diff
@@ -47,9 +47,6 @@
       form?.submit();
     }
   }
-  function returnCallback(callback: any) {
```
Comment (cerebrl) [inline src/lib/journey/stages/registration.svelte:50]: Not used.

File: src/lib/widget/_utilities/api.utilities.ts [hunk 4]
```diff
@@ -38,9 +43,6 @@ export function widgetApiFactory(componentApi: ReturnType<typeof _componentApi>)
     journeyStore.reset();
     oauthStore.reset();
     userStore.reset();
-
-    // Fetch fresh journey step
-    journey().start();
```
Comment (cerebrl) [inline src/lib/widget/_utilities/api.utilities.ts:43]: This wasn't needed, and resulted in a journey rendered that may not be relevant for the user that just logged out.

File: src/routes/e2e/widget/modal/+page.svelte [hunk 5]
```diff
@@ -61,8 +61,27 @@
...
+        },
+        login: {
+          journey: 'TEST_Login',
+          match: ['#/service/TEST_Login', '?journey', '?journey=TEST_Login', '#/service/Login', '?journey', '?journey=Login'],
+        },
+        register: {
+          journey: 'TEST_Registration',
+          match: ['#/service/TEST_Registration', '?journey=TEST_Registration', '#/service/Registration', '?journey=Registration'],
+        },
+      },
```
Comment (cerebrl) [inline src/routes/e2e/widget/modal/+page.svelte:84]: Testing the journey config.

File: src/lib/journey/journey.store.ts [hunk 6]
```diff
@@ -27,6 +27,15 @@ function initializeStack(initOptions?: StepOptions) {
 
   // Assign to exported variable (see bottom of file)
   stack = {
+    latest: async (): Promise<StepOptions> => {
```
[the remark this answers is not in the document]
Comment (ryanbas21) [reply src/lib/journey/journey.store.ts:30]: Not that this matters, but I think this is normally called `peek`.

**Pull request merged**